src/main/groovy/com/lesfurets/jenkins/unit/BasePipelineTest.groovy

```diff
@@ -122,6 +122,10 @@ abstract class BasePipelineTest {
         helper.registerAllowedMethod("choice", [Map])
         helper.registerAllowedMethod('cifsPublisher', [Map], {true})
         helper.registerAllowedMethod('cleanWs')
+        helper.registerAllowedMethod("container", [String, Closure], { String path, Closure c ->
+            c.delegate = delegate
+            helper.callClosure(c)
+        })
```

Comment:
```suggestion
        helper.registerAllowedMethod("container", [Map, Closure], { Map args, Closure c ->
            c.delegate = delegate
            helper.callClosure(c)
        })
        helper.registerAllowedMethod("container", [String, Closure], { String path, Closure c ->
            c.delegate = delegate
            helper.callClosure(c)
        })
```

I'm not terribly familiar with the Kubernetes plugin, but it seems that it supports taking both a `String` and a `Map`. It would be nice if this PR supported both.

Reply:
A have only seen container used like this so far. Do you have an example? 

As far as i see here it just refers to a previously defined name:
https://github.com/jenkinsci/kubernetes-plugin

ContainerTemplate is a different story but i thought that was more of an agent like thing to run under.

Reply:
https://www.jenkins.io/doc/pipeline/steps/kubernetes/ 
I see you are right. Couldn't find it at first. But it allows a name with shell combo.